File: robocode2-gui/src/main/kotlin/net/robocode2/gui/model/event/BulletHitBulletEvent.kt
package net.robocode2.gui.model.event

import net.robocode2.gui.model.comm.Content
import net.robocode2.gui.model.comm.ContentType

class BulletHitBulletEvent(
        val bullet: BulletState,
        val hitBullet: BulletState
) : Content(type = ContentType.BULLET_HIT_BULLET_EVENT.type)

File: robocode-tankroyale-bot-api-csharp/src/events/BulletHitBulletEvent.cs
using Newtonsoft.Json;

namespace Robocode.TankRoyale.BotApi
{
  /// <summary>
  /// Event occurring when a bullet has collided with another bullet.
  /// </summary>
  public sealed class BulletHitBulletEvent : Event
  {
    /// <summary>Bullet that hit another bullet.<summary>
    public BulletState Bullet { get; }

    /// <summary>The other bullet that was hit by the bullet.<summary>
    public BulletState HitBullet { get; }

    /// <summary>
    /// Constructor.
    /// </summary>
    /// <param name="turnNumber">Turn number.</param>
    /// <param name="bullet">Bullet that hit another bullet.</param>
    /// <param name="hitBullet">The other bullet that was hit by the bullet.</param>
    /// <returns></returns>
    [JsonConstructor]
    public BulletHitBulletEvent(int turnNumber, BulletState bullet, BulletState hitBullet) : base(turnNumber) =>
      (Bullet, HitBullet) = (bullet, hitBullet);
  }
}